### PR TITLE
Add optional grouped parsers description to HfArgumentParser

### PR DESCRIPTION
**Pull request by peteriz**
# What does this PR do?

This PR adds optional grouping to the argument parser of `HfArgumentParser` when multiple dataclasses are used (with different sub-grouped parameters, such as optimizer setup, model config, etc.) so that the displayed `-h` will print multiple grouped arguments in a more semantically organized way.
Uses optional attribute `_argument_group_name=<some string>` in the dataclass. If exists in the dataclass a sub parser will be used instead of the root `HfArgumentParser`.

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [X] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/master/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [X] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/master/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/master/docs#writing-source-documentation).
- [X] Did you write any new necessary tests?

@peteriz: Updated docstring inline

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

@sgugger

### CODE REVIEW

File: src/transformers/hf_argparser.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ class HfArgumentParser(ArgumentParser):
 
     The class is designed to play well with the native argparse. In particular, you can add more (non-dataclass backed)
     arguments to the parser after initialization and you'll get the output back after parsing as an additional
-    namespace.
+    namespace. Optional: To create sub argument groups use the `_argument_group_name` attribute in the dataclass.
```
Comment (sgugger) [inline src/transformers/hf_argparser.py:49]:
```suggestion
    namespace.
    
    Optional: To create sub argument groups use the `_argument_group_name` attribute in the dataclass.
```

**Review by sgugger — APPROVED** [review]
Thanks for adding this!

**Review by LysandreJik — APPROVED** [review]
Cool!

**Comment by peteriz** [comment]
Thanks!

**Pull request merged**